packages/@lwc/synthetic-shadow/src/faux-shadow/focus.ts

```diff
@@ -104,14 +105,23 @@ const focusableTagNames = {
 // This function based on https://allyjs.io/data-tables/focusable.html
 // It won't catch everything, but should be good enough
 // There are a lot of edge cases here that we can't realistically handle
-function isFocusable(element: HTMLElement): boolean {
-    const tagName = tagNameGetter.call(element);
-    return (
+function isFocusable(element: HTMLElement, hostElement: HTMLElement): boolean {
+    const focusable =
         isVisible(element) &&
         (hasFocusableTabIndex(element) ||
             hasAttribute.call(element, 'contenteditable') ||
-            hasOwnProperty.call(focusableTagNames, tagName))
-    );
+            hasOwnProperty.call(focusableTagNames, tagNameGetter.call(element)));
+
+    if (isTrue(focusable)) {
+        return true;
+    }
+
+    const nextElement =
```

Comment:
I'm not sure about this condition... I don't understand it.

`element.parentNode === element.getRootNode()` means that you're a top level element of the shadow. Maybe the name `nextElement` is what confuses me.

Reply:
We want to keep bubbling up through any shadow boundaries until we hit the custom element of the component that is delegating focus. I can change that check to something else if there's a better way to detect a shadow root.